src/m365/planner/commands/bucket/bucket-get.ts

```diff
@@ -0,0 +1,215 @@
+    if (args.options.id) {
+      return Promise.resolve(args.options.id);
+    }
+
+    return this
+      .getPlanId(args)
+      .then((planId: string) => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans/${planId}/buckets`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: { id: string; name: string; }[] }>(requestOptions);
+      })
+      .then(response => {
+        const bucket: { id: string; name: string; } | undefined = response.value.find(val => val.name === args.options.title);
+
+        if (!bucket) {
+          return Promise.reject(`The specified bucket does not exist`);
+        }
+
+        return Promise.resolve(bucket.id);
+      });     
+  }
+
+  private getPlanId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve(args.options.planId);
+    }
+
+    return this
+      .getGroupId(args)
+      .then((groupId: string) => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans?$filter=(owner eq '${groupId}')`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: { id: string; title: string; }[] }>(requestOptions);
+      })
+      .then(response => {
+        const plan: { id: string; title: string; } | undefined = response.value.find(val => val.title === args.options.planName);
+
+        if (!plan) {
+          return Promise.reject(`The specified plan does not exist`);
+        }
+
+        return Promise.resolve(plan.id);
+      });
+  }
+
+  private async getBucketById(id: string): Promise<PlannerBucket> {
+    const requestOptions: AxiosRequestConfig = {
+      url: `${this.resource}/v1.0/planner/buckets/${encodeURIComponent(id)}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'      
+    };
+
+    return request.get<PlannerBucket>(requestOptions);    
+  }
+
+  private getGroupId(args: CommandArgs): Promise<string> {
+    if (args.options.ownerGroupId) {
+      return Promise.resolve(args.options.ownerGroupId);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(args.options.ownerGroupName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: { id: string; }[] }>(requestOptions)
+      .then(response => {
+        const group: { id: string; } | undefined = response.value[0];
+
+        if (!group) {
+          return Promise.reject(`The specified owner group does not exist`);
+        }
+
+        return Promise.resolve(group.id);
```

Comment:
Let us handle the multiple group use case with the same displayname here. The user should be able to differentiate by id if there are more than one